ecl/eclccserver/eclccserver.cpp

```diff
@@ -235,6 +235,45 @@ static bool getHomeFolder(StringBuffer & homepath)
     return true;
 }
 
+static bool guardGitUpdates = false;
+static StringBuffer gitLockKey;
+static void configGitLock()
+{
+    Owned<IPropertyTree> config = getComponentConfig();
+    if (config->getPropBool("@enableEclccDali", true))
+    {
+        if (config->getPropBool("@guardGitUpdates", true))
+        {
+            if (isContainerized())
+            {
+                //Containerized: each git plane needs to be protected independently
+                gitLockKey.append(config->queryProp("@gitPlane"));
+            }
+            else
+            {
+                //Bare metal - git repos are fetched locally, so protect per host-ip
+                const char * hostname = GetCachedHostName();
+                if (hostname)
+                {
+                    gitLockKey.append("host");
+
+                    for (const byte * cur = (const byte *)hostname; *cur; cur++)
+                    {
+                        //remove '.' and other unsupported characters from the key name
+                        if (isalnum(*cur))
+                            gitLockKey.append(*cur);
+                        else
+                            gitLockKey.append("_");
+                    }
+                }
+            }
+
+            if (!gitLockKey.isEmpty())
+                guardGitUpdates = true;
```

Comment:
trivial: any point to non-empty gitLockKey and a boolean? i.e. could just test if (!gitLockKey.isEmpty()) below:

Reply:
Could do, but it is arguably slightly easier to read the code below.  I'm not sure it is worth changing.